privval/tcp_test.go

```diff
@@ -153,9 +146,9 @@ func TestSocketPVDeadline(t *testing.T) {
 	go func(sc *TCPVal) {
 		defer close(listenc)
 
-		require.NoError(t, sc.Start())
+		assert.Equal(t, sc.Start().(cmn.Error).Data(), ErrConnTimeout)
```

Comment:
Why is this supposed to error now?

Reply:
Because `OnStart` of `TCPVal` and `ICPVal` will call `NewRemoteSignerClient` which will request the public key:
https://github.com/tendermint/tendermint/blob/f9a564cfc37f7a7f6c2d2a19de72a73a71523ac5/privval/remote_signer.go#L29-L35

Reply:
This will time out (err) when calling Start(). Previously this error was supposed to happen after start: https://github.com/tendermint/tendermint/pull/2948/files#diff-f22985f984c7cb5c7d4f0702ea6325beL179